cpp/src/lists/drop_list_duplicates.cu

```diff
@@ -0,0 +1,291 @@
+                    });
+}
+/**
+ * @copydoc cudf::lists::drop_list_duplicates
+ *
+ * @param stream CUDA stream used for device memory operations and kernel launches
+ */
+std::unique_ptr<column> drop_list_duplicates(lists_column_view const& lists_column,
+                                             null_equality nulls_equal,
+                                             rmm::cuda_stream_view stream,
+                                             rmm::mr::device_memory_resource* mr)
+{
+  if (lists_column.is_empty()) return cudf::empty_like(lists_column.parent());
+  if (cudf::is_nested(lists_column.child().type())) {
+    CUDF_FAIL("Nested types are not supported in drop_list_duplicates.");
+  }
+
+  // Call segmented sort on the list elements
+  auto const sorted_lists = lists::sort_lists(lists_column, order::ASCENDING, null_order::AFTER);
+
+  // Flatten all entries (depth = 1) of the lists column
+  auto const all_lists_entries = lists_column_view(sorted_lists->view()).get_sliced_child(stream);
+
+  // Generate a 0-based offset column
+  auto lists_offsets = detail::generate_clean_offsets(lists_column, stream, mr);
+
+  // Generate a mapping from list entries to offsets of the lists containing those entries
+  auto const entries_list_offsets =
+    detail::generate_entry_list_offsets(all_lists_entries.size(), lists_offsets->view(), stream);
+
+  // Copy non-duplicated entries (along with their list offsets) to new arrays
+  auto unique_entries_and_list_offsets =
+    all_lists_entries.has_nulls()
+      ? detail::get_unique_entries_and_list_offsets<true>(
+          all_lists_entries, entries_list_offsets->view(), nulls_equal, stream, mr)
+      : detail::get_unique_entries_and_list_offsets<false>(
+          all_lists_entries, entries_list_offsets->view(), nulls_equal, stream, mr);
+
+  // Generate offsets for the new lists column
+  detail::generate_offsets(unique_entries_and_list_offsets.front()->size(),
+                           unique_entries_and_list_offsets.back()->view(),
+                           lists_offsets->mutable_view(),
+                           stream);
+
+  // Construct a new lists column without duplicated entries
+  return make_lists_column(lists_column.size(),
+                           std::move(lists_offsets),
+                           std::move(unique_entries_and_list_offsets.front()),
+                           lists_column.null_count(),
+                           cudf::detail::copy_bitmask(lists_column.parent(), stream, mr));
+}
+
+}  // anonymous namespace
+}  // namespace detail
+
+/**
+ * @copydoc cudf::lists::drop_list_duplicates
+ */
+std::unique_ptr<column> drop_list_duplicates(lists_column_view const& lists_column,
+                                             null_equality nulls_equal,
+                                             rmm::mr::device_memory_resource* mr)
+{
+  return detail::drop_list_duplicates(lists_column, nulls_equal, rmm::cuda_stream_default, mr);
```

Comment:
I'm not sure if we need to add `CUDF_FUNC_RANGE()` here?

Reply:
Yes, all external compute APIs should have `CUDF_FUN_RANGE` so they are annotated with NVTX.